File: cwe_examples/temporal_safety.cpp
class Bob {
    int *n;
public:
    Bob(int x) {
        this->n = new int[x];
    }
    ~Bob() {
        delete[] this->n;
    }
    int* get_n() {
        return this->n;
    }
};
int main() {
    Bob *b = new Bob(5);
    int *xs = b->get_n();
    delete b;
    printf("%d", xs[0]);
}

File: cwe_examples/temporal_safety.rs
struct Bob {
    n: Vec<i32>,
}
impl Bob {
    fn new() -> Bob {
        Bob { n: Vec::new() }
    }

    fn get_n(&self) -> &Vec<i32> {
        &self.n
    }
}
fn main() {
    let b = Bob::new();
    let xs = b.get_n();
    drop(b);
    println!("{:?}", xs[0]);
}
